platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -1030,3 +1030,165 @@ def get_sfp(self, index):
+            of modular-chassis.
+            For SmartSwitch platforms this should return True even if they are
+            fixed-platforms, as they are treated like a modular chassis as the
+            DPU cards are treated like line-cards of a modular-chassis.
+        """
+        return True
+
+    ##############################################
+    # Module methods
+    ##############################################
+    def initialize_single_module(self, index):
+        count = self.get_num_modules()
+        if index < count:
+            if not self._module_list:
+                self._module_list = [None] * count
+
+            if not self._module_list[index]:
+                from .module import DpuModule
+                self._module_list[index] = DpuModule(index + 1)
+                self.module_name_index_map[self._module_list[index].get_name()] = index
+                self.module_initialized_count += 1
+
+    def initialize_modules(self):
+        count = self.get_num_modules()
+        for index in range(count):
+            self.initialize_single_module(index=index)
+
+    def get_num_modules(self):
+        """
+        Retrieves the number of modules available on this chassis
+        On a SmarSwitch chassis this includes the number of DPUs.
+
+        Returns:
+            An integer, the number of modules available on this chassis
+        """
+        return DeviceDataManager.get_dpu_count()
+
+    def get_all_modules(self):
+        """
+        Retrieves all modules available on this chassis. On a SmarSwitch
+        chassis this includes the number of DPUs.
+
+        Returns:
+            A list of objects derived from ModuleBase representing all
+            modules available on this chassis
+        """
+        self.initialize_modules()
+        return self._module_list
+
+    def get_module(self, index):
+        """
+        Retrieves module represented by (0-based) index <index>
+        On a SmartSwitch index:0 will fetch switch, index:1 will fetch
+        DPU0 and so on
+
+        Args:
+            index: An integer, the index (0-based) of the module to
+            retrieve
+
+        Returns:
+            An object dervied from ModuleBase representing the specified
+            module
+        """
+        self.initialize_single_module(index)
+        return super(SmartSwitchChassis, self).get_module(index)
+
+    def get_module_index(self, module_name):
+        """
+        Retrieves module index from the module name
+
+        Args:
+            module_name: A string, prefixed by SUPERVISOR, LINE-CARD or FABRIC-CARD
+            Ex. SUPERVISOR0, LINE-CARD1, FABRIC-CARD5
+            SmartSwitch Example: SWITCH, DPU1, DPU2 ... DPUX
+
+        Returns:
+            An integer, the index of the ModuleBase object in the module_list
+        """
+        # TODO: Confirm whether Switch is module or not
+        ret_val = -1
+        try:
```

Comment:
You don't need try/except here. If the module name is wrong let __get__ method throw an exeption

Reply:
Removed